### PR TITLE
[PLAY-1581] Height, minHeight, maxHeight for Card, Flex, FlexItem, & Dialog

### PR DESCRIPTION
**Pull request by jasperfurniss**
**What does this PR do?** A clear and concise description with your runway ticket url.
https://runway.powerhrg.com/backlog_items/PLAY-1581

In order to achieve the height functionality that the story demanded, we introduced a new utility function for handling specific global properties that require _dynamic_ values (any px, %, or vh, etc). This has been on our Todo list for a looooong time, and I'm so happy we were able to knock out the work here in this story! 

The work in this PR adds the following props for react: **minHeight, maxHeight, height** for React
The work in this PR adds the following props for rails: **min_height, max_height, height** for Rails

For now, these are applied to a few of our utility kits: Card, Flex, FlexItem, Dialog

We have a path forward to apply to all other kits, but for now this work unblocks the Template work for the PB-In-Nitro team, without shipping changes to every kit which would move this to a much higher risk-level. 

**Screenshots:** Screenshots to visualize your addition/change

**How to test?** Steps to confirm the desired behavior:
1. Go to dialog, flex, flex_item, or card.
2. Add a height prop with a value. 
3. See your styles get applied.

#### Checklist:
- [X] **LABELS** Add a label: `enhancement`, `bug`, `improvement`, `new kit`, `deprecated`, or `breaking`. See [Changelog & Labels](https://github.com/powerhome/playbook/wiki/Changelog-&-Labels) for details.
- [X] **DEPLOY** I have added the `milano` label to show I'm ready for a review.
- [X] **TESTS** I have added test coverage to my code.

### CODE REVIEW

File: playbook/app/pb_kits/playbook/pb_flex/flex_item.rb [hunk 1]
```diff
@@ -20,8 +20,11 @@ def classname
         generate_classname("pb_flex_item_kit", fixed_size_class, grow_class, shrink_class, display_flex_class) + align_self_class
       end
 
-      def style_value
-        "flex-basis: #{fixed_size};" if fixed_size.present?
+      def inline_styles
+        styles = []
+        styles << "flex-basis: #{fixed_size};" if fixed_size.present?
+        styles << "height: #{height};" if height.present?
```
[the remark this answers is not in the document]
Comment (jasperfurniss) [reply playbook/app/pb_kits/playbook/pb_flex/flex_item.rb:26]: Good catch! Thanks @kangaree ! I'll add.

Comment (jasperfurniss) [inline playbook/app/pb_kits/playbook/pb_flex/flex_item.rb:26]:
```suggestion
      def inline_styles
        styles = []
        styles << "flex-basis: #{fixed_size};" if fixed_size.present?
        styles << "height: #{height};" if height.present?
        styles << "min-height: #{min_height};" if min_height.present?
        styles << "man-height: #{max_height};" if max_height.present?
```

**Review by nidaqg — APPROVED** [review]

**Comment by jasperfurniss** [comment]
> @jasperfurniss
> 
> Since we're adding these inline props to `kit_base.rb`, doesn't that mean we're adding these props to all Ruby kits?
> 
> Screenshot 2024-10-15 at 9 04 12 AM
> 
> Otherwise, looks really good to me! Clean, effective, and simple.
> 
> I've also got a few questions / comments which I've added.

@kangaree We are only adding these to kits that use pb_content_tag. But none of it should be getting used/applied yet. Did your screen shot show up automatically in a doc example? Or did you implement code to make that happen.

**Comment by kangaree** [comment]
> > @jasperfurniss
> > Since we're adding these inline props to `kit_base.rb`, doesn't that mean we're adding these props to all Ruby kits?
> > Screenshot 2024-10-15 at 9 04 12 AM
> > Otherwise, looks really good to me! Clean, effective, and simple.
> > I've also got a few questions / comments which I've added.
> 
> @kangaree We are only adding these to kits that use pb_content_tag. But none of it should be getting used/applied yet. Did your screen shot show up automatically in a doc example? Or did you implement code to make that happen.

I checked out this branch and in `playbook/app/pb_kits/playbook/pb_fixed_confirmation_toast/docs/_fixed_confirmation_toast_default.html.erb`, I made the following changes:

`<%= pb_rails("fixed_confirmation_toast", props: {
  text: "Items Successfully Moved!!!",
  status: "success",
  height: "500px",
})%>`

File: playbook/app/pb_kits/playbook/pb_flex/flex_item.html.erb [hunk 2]
```diff
@@ -1,8 +1,5 @@
-<%= content_tag(:div,
-    id: object.id,
-    data: object.data,
-    class: object.classname,
```
[the remark this answers is not in the document]
Comment (jasperfurniss) [reply playbook/app/pb_kits/playbook/pb_flex/flex_item.html.erb:4]: Correct! When we move to pb_content_tag, these are already taken care of in that method.

**Review by kangaree — APPROVED** [review]

**Comment by jasperfurniss** [comment]
> ```
> <%= pb_rails("fixed_confirmation_toast", props: {
>   text: "Items Successfully Moved!!!",
>   status: "success",
>   height: "500px",
> })%>
> ```

@kangaree Ah! Perfect! Then yes, you're ahead of the game, and testing the "beta" stuff that we aren't advertising for use yet. This will likely be a 1 of 2 ...or 1 of 3 story setup for this work. Thanks for testing that out!

**Review by nidaqg — APPROVED** [review]

**Pull request merged**